### PR TITLE
Add missing overview docs pages

### PR DESCRIPTION
**Pull request by petercrocker**
Added missing 4 overview pages:

- Data Transformation
- Version Control, CI and Checks
- generators & Design-Driven automation
- External integrations with automation frameworks (Ansible, Nornir)

### CODE REVIEW

File: docs/docs/media/design-versioning.excalidraw.svg
Comment (dgarros) [inline docs/docs/media/design-versioning.excalidraw.svg:1]: Looks like this diagram doesn't work well with a dark background, 
Screenshot 2024-07-17 at 05 57 20

**Review by dgarros — APPROVED** [review]
Looks good, appart from the design version diagram

File: docs/docs/media/design-versioning.excalidraw.svg
Comment (petercrocker) [reply docs/docs/media/design-versioning.excalidraw.svg:1]: Thanks, fixing now.

**Review by BeArchiTek — APPROVED** [review]
LGTM

**Pull request merged**